tests/gis_tests/gdal_tests/test_raster.py

```diff
@@ -467,3 +467,15 @@ def test_band_statistics_empty_band(self):
             'bands': [{'data': [0], 'nodata_value': 0}],
         })
         self.assertEqual(rsmem.bands[0].statistics(), (None, None, None, None))
+
+    def test_band_delete_nodata(self):
+        if GDAL_VERSION < (2, 1):
```

Comment:
How about checking for the `ValueError` on older versions? Would be nice to have a test for the `Nodata value must be numeric or None` error as well.